AFQ/tasks/segmentation.py

```diff
@@ -63,13 +64,9 @@ def segment(subses_dict, data_imap, mapping_imap,
         reg_template=reg_template,
         mapping=mapping_imap["mapping"])
 
-    if segmentation_params['return_idx']:
-        idx = {bundle: bundles[bundle]['idx'].tolist()
-               for bundle in bundle_dict}
-        bundles = {bundle: bundles[bundle]['sl']
-                   for bundle in bundle_dict}
+    seg_sft = aus.SegmentedSFT(bundles)
```

Comment:
I think that maybe there is an issue right here. These bundles are coming out of the segmentation in RASMM: 

https://github.com/yeatmanlab/pyAFQ/blob/cc3ee8aa03905a19a5b994cbdf011ac78539f47c/AFQ/segmentation.py#L815-L817

but the SegmentedSFT initializer assumes that they are in VOX:

https://github.com/yeatmanlab/pyAFQ/blob/cc3ee8aa03905a19a5b994cbdf011ac78539f47c/AFQ/utils/streamlines.py#L32. 

I believe that this was previously handled here: 

https://github.com/yeatmanlab/pyAFQ/blob/master/AFQ/utils/streamlines.py#L49

Reply:
Certainly something weird going on there.

I actually think that that line 815 that you linked to is a bug in master, because I think select_sl is in voxel space. I think this line in master:
https://github.com/yeatmanlab/pyAFQ/blob/master/AFQ/utils/streamlines.py#L49
does not actually perform any transformation, and that in this line in master:
https://github.com/yeatmanlab/pyAFQ/blob/4f7ce76e53fc9698b799f95a85586d3a213b44ff/AFQ/utils/streamlines.py#L51
Those same streamlines that are said to be RASMM space in 815 are said to be voxel before saving, without being transformed between those two lines. Thus when saved a transformation from voxel to RASMM is done, so master is not ultimately wrong, but line 815 is wrong and if you are working with the result of segmentation.segment you could run into problems.

Reply:
I think the problem in this PR only has to do with extracting profiles. In master, we keep streamlines in RASMM before extracting:
https://github.com/yeatmanlab/pyAFQ/blob/4f7ce76e53fc9698b799f95a85586d3a213b44ff/AFQ/tasks/segmentation.py#L313
and in my new segmentedSFT system, I was passing in VOX streamlines, causing the bug. 

I think I will address issues #480 and #616 next because I think the reason I did not catch this bug earlier is because stanford hardi is basically already in MNI space, so it was all working fine even though the spaces are wrong. Instead I have been using an HCP subject to test how this works. 

My plan to deal with this is to not assume segmentedSFT's sft is in vox or rasmm, in order to avoid unnecessary transformations. Instead, I will put the sft in the right space before I use it depending on the function. IE, in this case, when I call segmentedSFT.sft.to_rasmm() before extracting profiles, it works.

Reply:
Maybe one way to deal with this is to require a space input on the initialization of the SegmentedSFT object?

Reply:
Yes, this is one of the changes I made!